core/templates/dev/head/domain/exploration/SolutionObjectFactory.js

```diff
@@ -0,0 +1,125 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating new frontend instances of Solution
+ * domain objects.
+ */
+
+oppia.factory('SolutionObjectFactory', [
+  '$filter', 'oppiaHtmlEscaper',
+  function($filter, oppiaHtmlEscaper) {
+    var Solution = function(answerIsExclusive, correctAnswer, explanation) {
+      this.answerIsExclusive = answerIsExclusive;
+      this.correctAnswer = correctAnswer;
+      this.explanation = explanation;
+    };
+
+    Solution.prototype.toBackendDict = function() {
+      return {
+        answer_is_exclusive: this.answerIsExclusive,
+        correct_answer: this.correctAnswer,
+        explanation: this.explanation
+      };
+    };
+
+    Solution.createFromBackendDict = function(solutionBackendDict) {
+      return new Solution(
+        solutionBackendDict.answer_is_exclusive,
+        solutionBackendDict.correct_answer,
+        solutionBackendDict.explanation);
+    };
+
+    Solution.createNew = function(
+      answerIsExclusive, correctAnswer, explanation) {
+      return new Solution(answerIsExclusive, correctAnswer, explanation);
+    };
+
+    Solution.prototype.getSolutionSummary = function(interactionId, choices) {
+      console.log(this.correctAnswer);
+      console.log(choices);
+      var isExclusiveAnswer = (
+        this.answerIsExclusive ? 'Only' : 'One');
+      var correctAnswer = '';
+      if (interactionId === 'GraphInput') {
+        correctAnswer = '[Graph Object]';
+      } else if (interactionId === 'MultipleChoiceInput') {
+        correctAnswer = (
+          oppiaHtmlEscaper.objToEscapedJson(
+            choices[this.correctAnswer]
+              .label));
+      } else if (interactionId === 'MathExpressionInput') {
+        correctAnswer = this.correctAnswer.latex;
+      } else if (interactionId === 'CodeRepl' ||
+        interactionId === 'PencilCodeEditor') {
+        correctAnswer = this.correctAnswer.code;
+      } else if (interactionId === 'MusicNotesInput') {
+        correctAnswer = '[Music Notes Object]';
+      } else if (interactionId === 'ImageClickInput') {
+        correctAnswer = this.correctAnswer.clickedRegions;
+      } else if (interactionId === 'LogicProof') {
+        correctAnswer = this.correctAnswer.correct;
+      } else {
+        correctAnswer = (
+          oppiaHtmlEscaper.objToEscapedJson(this.correctAnswer));
+      }
+      var explanation = (
+        $filter('convertToPlainText')(this.explanation));
+      return (
+        '[' + isExclusiveAnswer + ' solution is ' + correctAnswer + '] ' +
+        explanation);
+    };
+
+    Solution.prototype.setCorrectAnswer = function (correctAnswer) {
+      this.correctAnswer = correctAnswer;
+    };
+
+    Solution.prototype.getCorrectAnswerHtml = function (objectType) {
+      if (objectType === 'CodeString') {
```

Comment:
They are used for different purposes, first one returns a full summary of the solution for the solution editor and the other returns the html for the correct answer alone -- this is directly bound to a div in the frontend.